test/store_test.mjs

```diff
@@ -38,50 +38,91 @@ test("if message passes constraint", async (t) => {
   await rm("dbtestA", { recursive: true });
 });
 
+test("returns leaves in ascending key order, small set", async (t) => {
+  env.DATA_DIR = "dbtestA";
+  const trieA = await store.create();
+  await trieA.put(Buffer.from("0c", "hex"), encode({ num: 0xc }));
+  await trieA.put(Buffer.from("05", "hex"), encode({ num: 0x5 }));
+  await trieA.put(Buffer.from("0a", "hex"), encode({ num: 0xa }));
+
+  const leaves = await store.leaves(trieA);
+  t.is(leaves.length, 3);
+  t.deepEqual([{ num: 0x5 }, { num: 0xa }, { num: 0xc }], leaves);
+
+  await rm("dbtestA", { recursive: true });
+});
+
+test("returns leaves in ascending key order, large set", async (t) => {
```

Comment:
This test ensures we get correct behaviour when the tree is more than 1 level deep.